### PR TITLE
added function - eachMinuteOfInterval

### PR DESCRIPTION
**Pull request by vitorfrs-dev**
Added the function eachMinuteOfInterval, i did the building process on my machine just like the documentation tells in the Contributing Guide, but the build process have change a bunch of files and the commit would be giant, this is why i commiting only 3 files and not the builded version.

This feature was requested on this issue: #2009

### CODE REVIEW

File: src/eachMinuteOfInterval/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  step?: number
+}
+
+/**
+ * @name eachMinuteOfInterval
+ * @category Interval Helpers
+ * @summary Return the array of hours within the specified time interval.
+ *
+ * @description
+ * Return the array of minutes within the specified time interval.
```
Comment (abnersajr) [inline src/eachMinuteOfInterval/index.ts:21]: I'm not sure if the description and summary are talking about the same function.

File: src/eachMinuteOfInterval/index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * @param {Interval} interval - the interval. See [Interval]{@link docs/types/Interval}
+ * @param {Object} [options] - an object with options.
+ * @param {Number} [options.step=1] - the step to increment by. Th starts of hours from the hour of the interval start to the hour of the interval end
```
Comment (abnersajr) [inline src/eachMinuteOfInterval/index.ts:25]: I think it's a typo here
```suggestion
 * @param {Number} [options.step=1] - the step to increment by. The starts of hours from the hour of the interval start to the hour of the interval end
```

File: src/eachMinuteOfInterval/index.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * @throws {TypeError} 1 argument requie value should be more than 1.
+ * @returns {Date[]} the array withred
+ * @throws {RangeError} `options.step` must be a number greater than 1
```
Comment (abnersajr) [inline src/eachMinuteOfInterval/index.ts:28]:
```suggestion
 * @throws {RangeError} `options.step` must be a number equal or greater than 1
```

File: src/eachMinuteOfInterval/index.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * @throws {RangeError} The start of an interval cannot be after its end
+ * @throws {RangeError} Date in interval cannot be `Invalid Dzate`
```
Comment (abnersajr) [inline src/eachMinuteOfInterval/index.ts:30]:
```suggestion
 * @throws {RangeError} Date in interval cannot be `Invalid Date`
```

**Review by abnersajr — CHANGES_REQUESTED** [review]

**Comment by kossnocorp** [comment]
> i did the building process on my machine just like the documentation tells in the Contributing Guide, but the build process have change a bunch of files and the commit would be giant, this is why i commiting only 3 files and not the builded version.

Good call, thank you!

File: src/eachMinuteOfInterval/index.ts (hunk 1 above)
Comment (kossnocorp) [reply src/eachMinuteOfInterval/index.ts]: In simple functions, they duplicate each other, but often the description contains much more info.

File: src/eachMinuteOfInterval/index.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import addMinutes from '../addMinutes/index'
+import toDate from '../toDate/index'
+import startOfMinute from '../startOfMinute/index'
+import requiredArgs from '../_lib/requiredArgs/index'
+
+interface Interval {
+  start: Date | number;
+  end: Date | number;
+}
```
Comment (kossnocorp) [inline src/eachMinuteOfInterval/index.ts:9]: Please import it from `../types`

File: src/eachMinuteOfInterval/index.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+interface Options {
+  step?: number
+}
```
Comment (kossnocorp) [inline src/eachMinuteOfInterval/index.ts:13]: Please rename it to `StepOptions` and move to `src/types.ts`

File: src/eachMinuteOfInterval/index.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+ * @returns {Date[]} the array withred
+ * @throws {RangeError} `options.step` must be a number equal or greater than 1
+ * @throws {RangeError} The start of an interval cannot be after its end
+ * @throws {RangeError} Date in interval cannot be `Invalid Date`
+ *
+ * @example
+ * // Each minute between 14 October 2020, 13:00 and 14 October 2020, 13:03
+ * const result = eachMinuteOfInterval({
+ *   start: new Date(2014, 9, 6, 12),
+ *   end: new Date(2014, 9, 6, 15)
```
Comment (kossnocorp) [inline src/eachMinuteOfInterval/index.ts:36]:
```suggestion
 *   start: new Date(2014, 9, 14, 13),
 *   end: new Date(2014, 9, 14, 13, 3)
```

File: src/eachMinuteOfInterval/index.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+): Date[] {
+  requiredArgs(1, arguments)
+
+  const startDate = startOfMinute(toDate(interval.start));
+  const endDate = startOfMinute(toDate(interval.end));
+
+  const startTime = startDate.getTime();
+  const endTime = endDate.getTime();
+
+  if(startTime >= endTime) {
```
Comment (kossnocorp) [inline src/eachMinuteOfInterval/index.ts:57]: While format the file using Prettier.

**Review by kossnocorp — CHANGES_REQUESTED** [review]
The code looks good, thank you a lot! Please address my comments and the PR is ready to go.

File: src/eachMinuteOfInterval/index.ts (hunk 6 above)
Comment (vitorfrs-dev) [reply src/eachMinuteOfInterval/index.ts]: I moved all the interfaces to types.ts

**Comment by RubenPonce** [comment]
Hey I really like the each day version of the function any chance we could get this one added?

**Comment by vitorfrs-dev** [comment]
@kossnocorp, I've seen that Travis is failing in the build process, and looks that this problem is happening in other pull requests as well. Do you know what might be causing this problem?

**Comment by kamranayub** [comment]
> @kossnocorp, I've seen that Travis is failing in the build process, and looks that this problem is happening in other pull requests as well. Do you know what might be causing this problem?

Travis basically shut off for all OSS projects 😞 Have to either request special credits or migrate to new CI. We moved to GitHub Actions ourselves but Circle CI is also an option.

**Comment by yaseenawazah** [comment]
This is really nice to have, is there something I can do to help get this code merged?

**Comment by tan75** [comment]
> This is really nice to have, is there something I can do to help get this code merged?

@yaseenawazah his has been merged via https://github.com/date-fns/date-fns/pull/2382

**Comment by tan75** [comment]
hi @vitorfrs-dev,
Thank you for your contribution!
Closing this PR as this has been merged via https://github.com/date-fns/date-fns/pull/2382

**Pull request closed without merging**